conan/tools/gnu/gnutoolchain.py

```diff
@@ -0,0 +1,280 @@
+    build_type_link_flags, \
+    libcxx_flags
+from conan.tools.env import Environment
+from conan.tools.gnu.get_gnu_triplet import _get_gnu_triplet
+from conan.tools.microsoft import VCVars, msvc_runtime_flag, unix_path, check_min_vs, is_msvc
+from conans.model.pkg_type import PackageType
+
+
+class GnuToolchain:
+    """
+    GnuToolchain generator.
+
+    Note: it's based on legacy AutotoolsToolchain but with a more modern and usable UX
+    """
+    def __init__(self, conanfile, namespace=None, prefix="/"):
+        """
+        :param conanfile: The current recipe object. Always use ``self``.
+        :param namespace: This argument avoids collisions when you have multiple toolchain calls in
+               the same recipe. By setting this argument, the *conanbuild.conf* file used to pass
+               information to the build helper will be named as *<namespace>_conanbuild.conf*. The default
+               value is ``None`` meaning that the name of the generated file is *conanbuild.conf*. This
+               namespace must be also set with the same value in the constructor of the Autotools build
+               helper so that it reads the information from the proper file.
+        :param prefix: Folder to use for ``--prefix`` argument ("/" by default).
+        """
+        raise_on_universal_arch(conanfile)
+        self._conanfile = conanfile
+        self._namespace = namespace
+        self._is_apple_system = is_apple_os(self._conanfile)
+        self._prefix = prefix
+
+        # Extra flags
+        self.extra_cxxflags = []
+        self.extra_cflags = []
+        self.extra_ldflags = []
+        self.extra_defines = []
+        # Extra environment definitions
+        self.extra_env = Environment()
+        # Defines
+        self.ndebug = None
+        build_type = self._conanfile.settings.get_safe("build_type")
+        if build_type in ['Release', 'RelWithDebInfo', 'MinSizeRel']:
+            self.ndebug = "NDEBUG"
+
+        # TODO: This is also covering compilers like Visual Studio, necessary to test it (&remove?)
+        self.build_type_flags = build_type_flags(self._conanfile.settings)
+        self.build_type_link_flags = build_type_link_flags(self._conanfile.settings)
+
+        self.cppstd = cppstd_flag(self._conanfile)
+        self.arch_flag = architecture_flag(self._conanfile.settings)
+        self.libcxx, self.gcc_cxx11_abi = libcxx_flags(self._conanfile)
+        self.fpic = self._conanfile.options.get_safe("fPIC")
+        self.msvc_runtime_flag = self._get_msvc_runtime_flag()
+        self.msvc_extra_flags = self._msvc_extra_flags()
+
+        # Cross build triplets
+        self.cross_build = {
+            "host": {"triplet": self._conanfile.conf.get("tools.gnu:host_triplet")},
+            "build": {"triplet": self._conanfile.conf.get("tools.gnu:build_triplet")}
+        }
+        is_cross_building = cross_building(self._conanfile)
```

Comment:
This is a bit confusing. The field is ``cross_build``, but having them different doesn't imply it is cross-building, that is a different check. So you could have the ``self.cross_build`` assigned and with different values, and still  not be cross-build? Then it should be renamed to ``self.host_triplet`` and ``self.build_triplet``?